### PR TITLE
Adding support for Forth code Highlight - issue #182

### PR DESCRIPTION
**Pull request by 0x3508A**
This is my contribution towards adding Forth code highlighting support in the `qmarkdowntextedit` control.

Kindly review the same and let me know if any changes would be needed.

### CODE REVIEW

**Comment by Waqar144** [comment]
Too many formatting changes in the patch. Please use clang-format for formatting so that actual changes can be seen. I suspect you have been using whatever the default formatter of your editor is

**Comment by pbek** [comment]
Hehe, I just wanted to mention that. 😊

**Comment by 0x3508A** [comment]
Thanks a lot for responding. I thought it would take more time.
Sorry about the format, I just realized that.
Please give me some time I will try to understand and update the code.

**Comment by 0x3508A** [comment]
I have tried my best to keep the formatting intact.
Please verify.

File: qownlanguagedata.h [hunk 1]
```diff
@@ -27,22 +27,16 @@
  * builtin are the library functions
  * other can contain any other thing, for e.g, in cpp it contains the
  preprocessor
-
```
Comment (pbek) [inline qownlanguagedata.h:30]: You still have a lot of newline changes. 😉

**Comment by pbek** [comment]
If you remove all newline-changes we should be good. 😉  Do you also care to integrate this into QOwnNotes after it's merged?

**Comment by 0x3508A** [comment]
> If you remove all newline-changes we should be good. 😉 Do you also care to integrate this into QOwnNotes after it's merged?

Sure, I would love to take this forward. As a user of QOwnNotes, It would be really nice to see changes reflected.
Sorry about the new line part next time I will use section formatting and Patch submission.

If its ok, I got my MSYS2 setup correctly working on Windows. I can create a patch and submit it.
Unfortunately I don't know much of Qt framework. Only a little bit of C++, C and go.
It would be great if you can point me to some sources, I will learn and build code locally also.

**Comment by pbek** [comment]
It's just a small change like: <https://github.com/pbek/QOwnNotes/commit/882f9263549ff10d7a3c680426da3de9abdb032d#diff-be90f4b188402b1122c82a91718431b91e4b715fd9c7543b9c19bfe93eda9262>

As for building QOwnNotes: https://www.qownnotes.org/installation/building.html

**Comment by 0x3508A** [comment]
> It's just a small change like: [pbek/QOwnNotes@882f926#diff-be90f4b188402b1122c82a91718431b91e4b715fd9c7543b9c19bfe93eda9262](https://github.com/pbek/QOwnNotes/commit/882f9263549ff10d7a3c680426da3de9abdb032d#diff-be90f4b188402b1122c82a91718431b91e4b715fd9c7543b9c19bfe93eda9262)
> 
> As for building QOwnNotes: https://www.qownnotes.org/installation/building.html

Thanks for sharing.
I will go through the build setup. For now to add the Forth support, I will do the small modification as suggested.

Thanks for guiding me through this.

I have checked the CI output it looks like Qt installed failed.
Kindly let me know if I need to any thing at my end to help.

**Comment by pbek** [comment]
> I have checked the CI output it looks like Qt installed failed.

GitHub has some issues today...

**Comment by 0x3508A** [comment]
I have prepared the changes for the QOwnNotes.
For now I will stash them, will take a fresh fork by filing an issue on the QOwnNotes.
Hope this would prevent the issue of too many changes and newlines.

This time the checks have passed.

**Comment by pbek** [comment]
>  This time the checks have passed.

waiting for your whitespace changes...

**Comment by 0x3508A** [comment]
There is some problem with merging. I will close the pull request and re-do the changes.

**Pull request closed without merging**